spec/models/video_spec.rb

```diff
@@ -37,5 +38,120 @@
+      review = Fabricate(:review, video: video)
+      expect(video.ordered_reviews).to eq([review])
+    end
+    it "should return reviews in reverse chronological order" do
+      video = Video.first
+      r1 = Fabricate(:review, video: video, created_at: 3.days.ago)
+      r2 = Fabricate(:review, video: video, created_at: 1.days.ago)
+      r3 = Fabricate(:review, video: video, created_at: 1.years.ago)
+      expect(video.ordered_reviews).to eq([r2, r1, r3])
+    end
+  end
+  describe "count_reviews" do
+    it "should return 0 if there are no reviews" do
+      video = Fabricate(:video)
+      expect(video.count_reviews).to eq(0)
+    end
+    it "should return 1 if there is only one review" do
+      video = Fabricate(:video)
+      Fabricate(:review, video: video)
+      expect(video.count_reviews).to eq(1)
+    end
+    it "should return 5 if there are 5 reviews" do
+      video = Fabricate(:video)
+      5.times { Fabricate(:review, video: video)}
+      expect(video.count_reviews).to eq(5)
+    end
+  end
+  describe "sum_review_ratings" do
+    it "should return 0 if there are no reviews" do
+      video = Fabricate(:video)
+      expect(video.sum_review_ratings).to eq(0)
+    end
+    it "should return 4 if there is one review with a rating of 4" do
+      video = Fabricate(:video)
+      Fabricate(:review, video: video, rating: 4)
+      expect(video.sum_review_ratings).to eq(4)
+    end
+    it "should return 5 if there are two reviews with a rating of 2 and 3" do
+      video = Fabricate(:video)
+      [2, 3].each do |i|
+        Fabricate(:review, video: video, rating: i)
+      end
+      expect(video.sum_review_ratings).to eq(5)
+    end
+    it "should return 15 if there are 5 reviews with ratings of 1,2,3,4, and 5" do
+      video = Fabricate(:video)
+      (1..5).each do |i|
+        Fabricate(:review, video: video, rating: i)
+      end
+      expect(video.sum_review_ratings).to eq(15)
+    end
+  end
+  describe "average_rating_of_reviews" do 
+    it "returns nil if there are no reviews" do
+      video = Fabricate(:video)
+      expect(video.average_rating_of_reviews).to be_nil
+    end
+    it "returns 1.0 if there is only one review with a rating of 1" do
+      video = Fabricate(:video)
+      Fabricate(:review, video: video, rating: 1)
+      expect(video.average_rating_of_reviews).to eq(1.0)
+    end
+    it "returns 5.0 if there is only one review with a rating of 5" do
+      video = Fabricate(:video)
+      Fabricate(:review, video: video, rating: 5)
+      expect(video.average_rating_of_reviews).to eq(5.0)
+    end
+    it "returns 5.0 if there are 10 reviews with a rating of 5 each" do
+      video = Fabricate(:video)
+      10.times {Fabricate(:review, video: video, rating: 5)}
+      expect(video.average_rating_of_reviews).to eq(5.0)
+    end
+    it "returns 1.0 if there are 20 reviews with a rating of 1 each" do
+      video = Fabricate(:video)
+      10.times {Fabricate(:review, video: video, rating: 1)}
```

Comment:
Your spec describes a scenario where there are 20 reviews but there are actually only 10 here. Make sure your specs and descriptions match. Also, 10 objects is a lot of overhead to fabricate every time your spec runs. Over time you will want your specs to run faster, this is one place to cut time pretty easily. Use 3 instead of 10 or 20 and you will save time exponentially. Also, if you use `Fabricate.build` it doesn't actually hit the db and you save some overhead as well, since they will only be created in memory.